File: c++/bubbleSort.cc
#include <vector>

// Sorts array of type T in ascending order
template <typename T> void bubbleSort(std::vector<T>& v) {
	int len = v.size();
	T temp;
	bool changedHuh = true;	

	for (typename std::vector<T>::iterator i = v.begin(); i != v.end(); ++i) {
		changedHuh = false;

		for (typename std::vector<T>::iterator j = v.begin() - 1; j != v.end(); ++j) {
			if (*i < *(j+1)) {			// Checks if current is larger than next
				temp = *i;				// If so, swap
				*i = *(j+1);
				*(j+1) = temp;
				changedHuh = true;
			}
		}

		if (!changedHuh) break;
	}	
}

File: python/bubble_sort.py
#Sorts array in increasing order
def bubble_sort(a):
	sortHuh = False
	length = len(a)

	while not sortHuh:								#While not sorted
		sortHuh = True
		for i in range (0,length-1):				
			if (a[i] > a[i+1]):						#If current is larger than next
				a[i], a[i+1] = a[i+1], a[i]			#Swap
				sortHuh = False
